homeworks/17/instagram-api/Post.java

```diff
@@ -0,0 +1,173 @@
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user_id);
+            statement.setString(2, image);
+            statement.setString(3, description);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        String pathInfo = request.getPathInfo();
+        String[] split = pathInfo.split("/");
+        post_id = split[1];
+
+        if (split.length > 2) {
+            if (split[2] == "comments") {
+                if (split.length > 3) {
+                    comment_id = split[3];
+                }
+                try {
+                    String query = "INSERT INTO comments(post_id, description)" + " values (?, ?)";
+                    PreparedStatement statement = connection.prepareStatement(query);
+                    statement.setString(1, post_id);
+                    statement.setString(2, description);
+                    statement.executeUpdate();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+	}
+	
+    public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html");
+
+        String pathInfo = request.getPathInfo();
+        String[] split = pathInfo.split("/");
+        post_id = split[1];
+
+        String description = request.getParameter("description");
+
+        if (split[2] == "comments") {
+            comment_id = split[3];
+            try {
+                Class.forName("com.mysql.jdbc.Driver");
+                Connection connection = DriverManager.getConnection("jdbc:mysql://localhost:3306/instagram", "root", "root");
+                String query = "UPDATE comments SET description = ? WHERE comments_id = ?";
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setString(1, description);
+                statement.setString(2, comment_id);
+                statement.executeUpdate();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    
+    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html");
+
+        String pathInfo = request.getPathInfo();
+        String[] split = pathInfo.split("/");
+        post_id = split[1];
+
+        if (split[2] == "comments") {
+            comment_id = split[3];
+            try {
+                String query = "DELETE FROM comments WHERE comments_id = ?";
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setString(1, comment_id);
+				statement.executeUpdate();
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+        }
+    }
```

Comment:
You could have checked if the user is logged in before proceeding to execute the queries, but overall fine! :+1: